geotrek/maintenance/views.py

```diff
@@ -152,12 +153,14 @@ def get_form_kwargs(self):
         if kwargs['can_delete']:
             intervention = self.get_object()
             # Disallow deletion if this intervention is part of Suricate Workflow at the moment
-            not_workflow = not settings.SURICATE_WORKFLOW_ENABLED
-            is_report = intervention.target and intervention.target.__class__.__name__ == "Report"
-            report_is_closed = False
-            if is_report:
-                report_is_closed = (intervention.target.status.identifier == 'solved')
-            kwargs["can_delete"] = not_workflow or (not is_report) or report_is_closed
+            if not settings.SURICATE_WORKFLOW_ENABLED:
+                kwargs["can_delete"] = True
+            else:
+                is_report = intervention.target and isinstance(intervention.target, Report)
+                report_is_closed = False
+                if is_report:
+                    report_is_closed = (intervention.target.status.identifier == 'solved')
+                kwargs["can_delete"] = (not is_report) or report_is_closed
```

Comment:
Changed execution flow to prevent an exception than _can_ happen when workflow is disabled - `intervention.target.status.identifier` might throw exception if there is no status (unlikely, but technically possible)